assets/src/components/map-wrapper.js

```diff
@@ -0,0 +1,253 @@
+      return;
+    }
+
+    // Styling of the map marker
+    const iconStyle = new Style({
+      image: new Icon({
+        anchor: [0.5, 46],
+        anchorXUnits: "fraction",
+        anchorYUnits: "pixels",
+        src: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABsAAAAlCAMAAACEemK6AAABXFBMVEUAAAAAgP8AZv8AgP8Abf8AYP8Acf8XdP8Vav8Udv8Lb/8KcP8PbP8Pcf8Ocf8Mbv8Lb/8NcP8Nb/8Nbf8Mb/8Mbf8MbvsLbfsPb/sObvsObPwNcPwNbvwMbfwObfwNbvwNbv0Mbv0Ob/0Obf0Mbf0Mbv0Mbv0Obv0Nbv0Nbf0Nbv0Nbv0Mb/0Mbv0Mbf0Obv0Ob/0Obv0Nb/0Nbf0Nbv0Nbv0Nbv0Nbv0Mbv0Mb/1ppv4Nbvwvgv4NbvwNbvwNb/y81/8Nb/2jyf4Nbf0kfP0Nbv0Nbv2PvP45if4Nbv0dd/0Nbv0Nbv0Nbv0Nbv0Nbv0Nbv1SmP4Nbv0Nbv08iv0Nb/0Nbv0Nbv32+f8Rcf3u9f8Mbv281/4Nbv2cxP4Nbv17sP4Nbv0Nbv0Xdf1foP4Nbv0Nbv0Nbv0Nbv07if1Umf5cnf6Nu/6x0P7H3f/t9P/u9f/2+f/4+//////ZrbbCAAAAaHRSTlMABAUGBwgJCwwNFxkhIiQsLjk8PT4/QURFSElQUVRZZGZrbG58fX+Ai4yNjpGSk5SWl5iam5ydoKSmv8PExcfLztLT1NTW2Nna3Nzd3t/l5+jo7O3v8PLz8/T09ff4+Pn5+vv7+/z9/kqOLf8AAAABYktHRHNBCT3OAAABKklEQVQYGW3BhVYCQQAF0CeoWBjYid2Nio2Fiig22IGCz475/3Nk2RlY2LkXSnGbPxC5jwT8rcXI5RqIUYn2u2DRFKFVpBEZ3kfmindB6qSdF2k1t7R7rEVK4QF1wk4APdTrBgqOqHdYgAZmbO3c3J0xox4jVBaF4YPKEFYoLQvTO6UlhCmtCdMXpRCuKV0I0y+la8QoXQnTD6UoNiltCNMnpXXMUtr7E4a/V0o+tFDZ/RFCfL9RaYbjlMrl8cPzE5UTBzBBvXEAFXHqJCqRskCdeRiqk7RLVCFtjnYzMJVFmS9WDqmP+XqhOIPMtV2EjLoXWiU9sPDRahpWJSFm7ZcihydJJVmPPMNUBpHPsUpTwAkb9zkN525otNPQAa0pkpPQKxwLjjqR9Q+OaxIua1fKFgAAAABJRU5ErkJggg==",
+      }),
+    });
+
+    // Define feature array and apply styling
+    const features = [];
+    const featureIds = [];
+
+    resourceData.forEach((value) => {
+      const feature = new Feature({
+        geometry: new Point([value.coordinates.easting, value.coordinates.northing]),
+        name: value.name,
+        address: value.address,
+        city: value.city,
+      });
+
+      featureIds.push(value.id);
+
+      feature.setStyle(iconStyle);
+
+      features.push(feature);
+    });
+
+    const vLayer = new VectorLayer({
+      // Resource features
+      name: "Vector",
+      source: new VectorSource({
+        features,
+      }),
+    });
+
+    if (map) {
+      setCurrentFeatures(featureIds);
+
+      setVectorLayer(vLayer);
+    }
+    */
+  }, [resources, map]);
+
+  // useEffect(() => {
+  // Handles removing old layers and adding new ones
+  // if (map && vectorLayer) {
+  //   map
+  //     .getLayers()
+  //     .getArray()
+  //     .forEach((value) => {
+  //       // Loop vectorLayers and remove old
+  //       if (value.get("name") === "Vector") {
+  //         map.removeLayer(value);
+  //       }
+  //     });
+  //   map.addLayer(vectorLayer); // Add newly defined vectorLayer
+  // }
+  // }, [vectorLayer, map]);
+
+  useEffect(() => {
+    // Current map instances
+    const mapChildren = mapElement.current.children;
+
+    let mapAlreadyLoaded = false;
+
+    // Check if map is already loaded
+    Object.values(mapChildren).forEach((mapChild) => {
+      if (mapChild.className.indexOf("ol-viewport") > -1) {
+        mapAlreadyLoaded = true;
```

Comment:
In this case you should break iteration since the result will always be true after running this line.